src/geoops/macros.jl

```diff
@@ -66,19 +66,23 @@ _getcolexpr(expr) = :(getproperty(geotable, $(_nameexpr(expr))))
 function _colnames(expr)
   colnames = []
   if expr isa Expr
-    _colargs!(expr, colnames)
+    _colargs!(colnames, expr)
   end
   colnames
 end
 
-function _colargs!(expr, colnames)
-  start = Meta.isexpr(expr, [:call, :macrocall]) ? 2 : 1
-  for i in start:length(expr.args)
-    arg = expr.args[i]
-    if _iscolname(arg)
-      push!(colnames, arg)
-    elseif arg isa Expr
-      _colargs!(arg, colnames)
+function _colargs!(colnames, expr)
+  if Meta.isexpr(expr, :.) # handle expressions of the form obj.field
```

Comment:
I think this might be enough:
```suggestion
  if Meta.isexpr(expr, :., 2) && expr.args[2] isa QuoteNode # handle expressions of the form obj.field
```

Reply:
I think it makes sense to check this even though the `@transform` is not supposed to be used with manual broadcasts. Thanks for catching it.